packages/data/src/world/mq/dodongo_cavern_mq.yml

```diff
@@ -2,159 +2,279 @@
+  events:
+    DC_MQ_DEFEAT_DONDONGOS_AFTER_STAIRCASE: "soul_dodongo && can_damage" #Placeholder macro for defeating Dodongos
   locations:
-    "MQ Dodongo Cavern Compass Chest": "soul_dodongo"
+    "MQ Dodongo Cavern Compass Chest": "event(DC_MQ_DEFEAT_DONDONGOS_AFTER_STAIRCASE)"
     "MQ Dodongo Cavern Grass Compass Room 1": "can_cut_grass"
     "MQ Dodongo Cavern Grass Compass Room 2": "can_cut_grass"
     "MQ Dodongo Cavern Grass Compass Room 3": "can_cut_grass"
     "MQ Dodongo Cavern Grass Compass Room 4": "can_cut_grass"
-"Dodongo Cavern Torch Room":
+"Dodongo Cavern Main Room Upper Entrance Side":
   dungeon: DC
   exits:
-    "Dodongo Cavern After Staircase": "true"
-    "Dodongo Cavern Upper Ledges": "can_hookshot || has_hover_boots || (is_adult && trick(OOT_DC_JUMP))"
-    "Dodongo Cavern Room Before Upper Lizalfos": "can_use_sticks || has_fire"
-  locations:
-    "MQ Dodongo Cavern Larvae Room Chest": "soul_enemy(SOUL_ENEMY_GOHMA_LARVA) && (can_use_sticks || has_fire)"
-    "MQ Dodongo Cavern GS Larve Room": "gs && (can_use_sticks || has_fire)"
-    "MQ Dodongo Cavern Heart Vanilla Bomb Bag Room": "true"
-"Dodongo Cavern Upper Lizalfos":
+    "Dodongo Cavern Main Room Lower": "true"
+    "Dodongo Cavern Room After Upper Staircase": "true"
+    "Dodongo Cavern Main Room Upper Skull Side": "longshot_anywhere"
+    "Dodongo Cavern Vanilla Bomb Bag Room Lower": "true"
+"Dodongo Cavern Vanilla Bomb Bag Room Lower":
   dungeon: DC
   exits:
-    "Dodongo Cavern Upper Ledges": "soul_lizalfos_dinalfos && (can_use_sticks || has_weapon || can_use_slingshot)"
+    "Dodongo Cavern Main Room Upper Entrance Side": "true"
+    "Dodongo Cavern Room Before Upper Lizalfos": "can_use_sticks || has_fire"
+    "Dodongo Cavern Vanilla Bomb Bag Room Upper": "can_hookshot || has_hover_boots || (is_adult && trick(OOT_DC_JUMP)) || climb_anywhere"
+    "Dodongo Cavern MQ Larvae Room": "event(OPEN_MQ_DC_LARVAE_ROOM)"
+  events:
+    OPEN_MQ_DC_LARVAE_ROOM: "can_use_sticks || has_fire"
   locations:
-    "MQ Dodongo Cavern GS Upper Lizalfos": "gs && has_explosives_or_hammer"
-    "MQ Dodongo Cavern Pot Miniboss 1": "true"
-    "MQ Dodongo Cavern Pot Miniboss 2": "true"
-    "MQ Dodongo Cavern Pot Miniboss 3": "true"
-    "MQ Dodongo Cavern Pot Miniboss 4": "true"
-"Dodongo Cavern Upper Ledges":
+    "MQ Dodongo Cavern Heart Vanilla Bomb Bag Room": "true"
+    # Fails to load
+    "MQ Dodongo Cavern Grass Vanilla Bomb Bag Room": "can_cut_grass"
+"Dodongo Cavern Vanilla Bomb Bag Room Upper":
   dungeon: DC
   exits:
+    "Dodongo Cavern Vanilla Bomb Bag Room Lower": "true"
+    "Dodongo Cavern Main Room Upper Skull Side": "event(DC_MQ_SHORTCUT) || climb_anywhere"
     "Dodongo Cavern Room After Upper Lizalfos": "true"
-    "Dodongo Cavern Torch Room": "true"
   events:
-    "DC_MQ_SHORTCUT": "true"
+    DC_MQ_SHORTCUT: "has_bombflowers || has_fire || has_explosives_or_hammer"
```

Comment:
Instead of "has_fire", it should be "can_use_din || can_use_bow" as you do not need fire arrows specifically.

Reply:
Fixed.